src/pages/leeszaal/[readingRoomSlug]/index.tsx

```diff
@@ -88,10 +96,31 @@ const ReadingRoomPage: NextPage = () => {
 	 */
 
 	const { data: mediaResultInfo } = useGetMediaObjects(
-		{
-			query: (query.search || []).join(' '),
-			format: (query.format as ReadingRoomMediaType) || READING_ROOM_QUERY_PARAM_INIT.format,
-		},
+		[
+			{
+				field: MediaSearchFilterField.QUERY,
+				operator: MediaSearchOperator.CONTAINS,
+				value: query.search !== null ? query.search?.toString() : '',
+			},
+			{
+				field: MediaSearchFilterField.FORMAT,
+				operator: MediaSearchOperator.IS,
+				value: query.format || READING_ROOM_QUERY_PARAM_INIT.format,
+			},
+			...(query.advanced || [])
+				.map((item) => {
+					const values = (item.val || '').split(SEPARATOR);
+					const filters =
+						item.prop && item.op
+							? getMetadataSearchFilters(
+									item.prop as MetadataProp,
+									item.op as Operator
+							  )
+							: [];
+					return filters.map((filter, i) => ({ ...filter, value: values[i] }));
+				})
+				.reduce((prev, filters) => prev.concat(filters), []),
```

Comment:
you might be able to just use a .flatMap() instead of .map().reduce()

Reply:
Fixed in https://github.com/viaacode/hetarchief-client/commit/b2c853abdcbc58511d542fb9eaa669cb78f4ac78